src/BinaryParsers/PEBinary/ProgramDatabase/Pdb.cs

```diff
@@ -86,10 +86,18 @@ private void Init(string peOrPdbPath, string symbolPath, string localSymbolDirec
             }
             catch (COMException ce)
             {
-                throw new PdbException(ce)
+                var pdbException = new PdbException(ce)
                 {
                     LoadTrace = this.loadTrace?.ToString()
                 };
+
+                if (pdbException.ExceptionCode == DiaHresult.E_PDB_NOT_FOUND && peOrPdbPath.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))

```

Comment:
>pdbException.ExceptionCode [](http://example.com/codeflow?start=20&length=26)

Why are we entering this code path? A path with '.dll' should not arrive here. btw - if we did intend to handle things in the way you've proposed, we would need to handle '.exe' extension and '.sys'.

In general, however, this is a PDB class. it's not the job of the PDB class to correct the situation if someone upstream has tried to initialize it with something that isn't a PDB. So I'd tend to go examine where that is occurring and look at adding path processing there.

PDB loading/resolution is a very tricky topic and the logic has evolved considerably over time. We should consider investing more in testing here, perhaps with mocking to simulate all necessary conditions. Something to keep in mind.

Reply:
when we analyze with the following command:
`analyze "C:\Users\ednakamu\Desktop\binskim\*.dll"`, the value of peOrPdbPath is equal to the dll.

When that happens, we will try to load normally using the method `loadDataForExe`. But that sometimes fails with PDB_NOT_FOUND. In that case, we want to try again reading the pdb manually.

What I tested:
- multiple  combinations of the boolean that configure the reader
- with and without symbolserver

if thats the case, i could make the change in the PEBinary, adding this logic near the caught clause of the LoadPdb method.